bindings/netstandard/ElectionGuard/ElectionGuard.Decryption/Verify/VerifyElection.cs

```diff
@@ -0,0 +1,1087 @@
+    ///       where L is the contest selection limit. 
+    /// An election verifier must then confirm the following:
+    /// (6.A) The given values αi and βi are each in Zrp.
+    /// (6.B) The given values cj each satisfy 0 ≤ cj < 2^256.
+    /// (6.C) The given values vj are each in Zq for all 0 ≤ j ≤ L. 
+    /// (6.D) The equation c = (c0 + c1 + ··· + cL) mod q is satisfied.
+    /// </summary>
+    public static async Task<VerificationResult> VerifyVoteLimits(
+        ElectionConstants constants,
+        CiphertextElectionContext context,
+        Manifest manifest,
+        List<CiphertextBallot> ballots
+    )
+    {
+        var results = new List<VerificationResult>();
+        foreach (var ballot in ballots)
+        {
+            var contestResults = new List<VerificationResult>();
+            foreach (var contest in ballot.Contests)
+            {
+                var accumulation = new ElGamalCiphertext(Constants.ONE_MOD_P, Constants.ONE_MOD_P);
+                foreach (var selection in contest.Selections)
+                {
+                    accumulation = accumulation.Add(selection.Ciphertext);
+                }
+                // Verification 6.1
+                var consistentA = accumulation.Pad.Equals(contest.CiphertextAccumulation.Pad);
+                contestResults.Add(new VerificationResult(consistentA, $"- Verification 6.1: {contest.ObjectId} Consistent A"));
+
+                // Verification 6.2
+                var consistentB = accumulation.Data.Equals(contest.CiphertextAccumulation.Data);
+                contestResults.Add(new VerificationResult(consistentB, $"- Verification 6.2: {contest.ObjectId} Consistent B"));
+
+                // Verification 6.3 - 6.D
+                contestResults.Add(await VerifyContestRangeProof(constants, context, manifest, contest));
+
+                accumulation.Dispose();
+            }
+            results.Add(new VerificationResult($"    - Verification 6: Ballot {ballot.ObjectId}", contestResults));
+        }
+
+        return new VerificationResult("Verification 6 (Adherence to vote limits)", results);
+    }
+
+    private static Task<VerificationResult> VerifyContestRangeProof(
+        ElectionConstants constants,
+        CiphertextElectionContext context,
+        Manifest manifest,
+        CiphertextBallotContest contest
+    )
+    {
+        var results = new List<VerificationResult>();
+
+        // TODO: Issue #420 - implement range proofs for selection
+        // https://github.com/microsoft/electionguard-core2/issues/420
+        // 
+        // Once the ZeroKnowledgeProof cpp struct is exposed across the api surface
+        // reimplement this method by verifying all of the math is correct manually
+        // instead of calling RangedChaumPedersenProof.IsValid. 
+
+        // Verification 6.3 - aj = g^vj ·α ̄^cj mod p
+        // Verification 6.4 - bj = K^wj · β ̄^cj mod p
+        var temp_internal_check_is_valid = contest.Proof.IsValid(
```

Comment:
this check "works" as intended but because the ZeroKnowledgeProof struct is not propagated across the api surface we can't actually validate the internal objects without creating new objects, which i considered out of scope for this task.